Embedded/Arduino/Motor_controller/src/Hardware/Battery/BatteryHealth.cpp

```diff
@@ -14,95 +18,83 @@ int buzzerPulseCount = 0;
 unsigned long lastWarningTriggerTime = 0;
 
 /*Variabler för LastSignal Start funktionen*/
-
 // ska vara false från början
 bool lastSignalActive = false;
 unsigned long lastSignalStartTime = 0;
 
+// Variabler för att skicka batteri status till servern
+bool warningSignal = false;
+bool sleepModeSigna = false; 
+
 // Läser av det analoga värdet från A0 och räknar ut laddningen av batteriet
-float CalculateBatteryHealth(int analogPin, float Vref)
-{
+float CalculateBatteryHealth(int analogPin, float Vref) {
   // Läser av det analoga Pinnen
   int rawADC = analogRead(analogPin);
   // 0-1023 (10 bitars)
   float voltage = (rawADC * Vref) / 1023.0;
   float batteryVoltage = voltage * ((R1 + R2) / R2);
-
   // Ger batteri nivån
   return batteryVoltage;
 }
-
-int CheckBatteryLevel(float current_level)
-{
-  if (current_level <= BATTERY_WARNING_LEVEL && current_level > BATTERY_SHUTDOWN_LEVEL)
-  {
+// Den funktionen används i Arduinon för att kolla batteri nivån
+int CheckBatteryLevel(float current_level) {
+  if (current_level <= BATTERY_WARNING_LEVEL && current_level > BATTERY_MIN_VOLTAGE) {
     return Battery_Warning;
   }
-  else if (current_level <= BATTERY_SHUTDOWN_LEVEL)
-  {
+  else if (current_level <= BATTERY_MIN_VOLTAGE) {
     return Battery_Shutdown;
   }
-  else
-  {
+  else {
     return Battery_OK;
   }
 }
 
-void CheckBatteryAndWarn()
-{
+void CheckBatteryAndWarn() {
   // Current voltage of the battery
   float voltage = CalculateBatteryHealth(A0, VREF);
-
   // Battery Level
   int level = CheckBatteryLevel(voltage);
 
-  if (level == Battery_Shutdown)
-  {
+  if (level == Battery_Shutdown) {
     // last signal
     TriggerWarningSignal(true);
-
-    // shutdown()
+    // Skickar true till servern för att säga att systemet ska stängas av
+    sleepModeSigna = true;
   }
-  else if (level == Battery_Warning)
-  {
-    // First signal
+  else if (level == Battery_Warning) {
+    // Skickar true till servern för att säga att batteriet är lågt
+    warningSignal = true;
     // Millis gör att den tiden inte pausar allt annat utan kör i bakgrunden
     unsigned long currentTime = millis();
     unsigned long timeSinceLastWarning = currentTime - lastWarningTriggerTime;
     // Så att det triggas igång varje 30 Sekunder
-    if (timeSinceLastWarning >= WARNING_REMINDER)
-    {
+    if (timeSinceLastWarning >= WARNING_REMINDER) {
       TriggerWarningSignal(false);
       lastWarningTriggerTime = currentTime;
     }
   }
-
   // Just for testing
   Serial.print("Current Voltage: ");
   Serial.print(voltage, 2);
   Serial.println(" V");
 }
 
 // Bestämmer vilken varning signal som ska användas och sen sätter igång den
-void TriggerWarningSignal(bool level)
-{
+void TriggerWarningSignal(bool level) {
```

Comment:
Let's change to a more descriptive name, check if this is sufficient
```suggestion
void TriggerWarningSignal(bool isLastSignal) {
```